lib/cisco_node_utils/bridge_domain.rb

```diff
@@ -104,30 +104,51 @@ def self.bds
     # idempotency issue as system add command throws error if a bd is already
     # present in the system range.
     def create
-      sys_bds_array = BridgeDomain.bd_ids_to_array(system_bd_add)
+      sys_bds_array = BridgeDomain.bd_ids_to_array(system_bridge_domain)
       inp_bds_array = BridgeDomain.bd_ids_to_array(@bd_ids)
       if (inp_bds_array - sys_bds_array).any?
         add_bds = BridgeDomain.bd_list_to_string(inp_bds_array - sys_bds_array)
-        config_set('bridge_domain', 'system_bd_add', addbd: add_bds)
+        config_set('bridge_domain', 'system_bridge_domain', oper: 'add',
+                                                            bd:   add_bds)
       end
-      config_set('bridge_domain', 'create', crbd: @bd_ids)
+      config_set('bridge_domain', 'create', bd: @bd_ids)
     end
 
     def destroy
-      config_set('bridge_domain', 'destroy', delbd: @bd_ids, rembd: @bd_ids)
+      config_set('bridge_domain', 'destroy', bd: @bd_ids)
+      config_set('bridge_domain', 'system_bridge_domain', oper: 'remove',
+                                                          bd:   @bd_ids)
     end
 
     ########################################################
     #                      PROPERTIES                      #
     ########################################################
 
+    # Bridge-Domain name assigning case
+    # bridge-domain 100
+    #   name PepsiCo
+    def bd_name
+      res = config_get('bridge_domain', 'bd_name', bd: @bd_ids)
+      res.nil? ? default_bd_name : res
+    end
+
+    def bd_name=(str)
+      str = str.to_s
+      state = str.empty? ? 'no' : ''
+      config_set('bridge_domain', 'bd_name', bd: @bd_ids, state: state,
+                   name: str)
+    end
+
+    def default_bd_name
+      sprintf('Bridge-Domain%s', @bd_ids)
```

Comment:
We cannot use get default because, when we create a bd name is assigned automatically;
As below the bridge-domain 100 has a name assigned as Bridge-Domain100 without explicitly assigning one.

Hence doing it in this way.

n7k-venus(config)# system bridge-domain 100
n7k-venus(config)# bridge-domain 100
n7k-venus(config-bdomain)# exit
n7k-venus(config)# show bridge-domain 100

Bridge-domain 100  (0 ports in all)
Name:: Bridge-Domain100
 Administrative State: UP               Operational State: DOWN

n7k-venus(config)#